tests/testdata/codegen/any-resource/schema.json

```diff
@@ -0,0 +1,81 @@
+{
+  "version": "0.0.1",
+  "name": "example",
+  "types": {
+    "example:index:TypeWithResources": {
+      "properties": {
+        "plainResource": {
+          "$ref": "pulumi.json#/Resource",
+          "plain": true
+        },
+        "resource": {
+          "$ref": "pulumi.json#/Resource"
+        }
+      },
+      "required": [
+        "resource"
+      ],
+      "type": "object"
+    }
+  },
+  "resources": {
+    "example:index:ResourceWithResources": {
+      "properties": {
+        "resource": {
+          "$ref": "pulumi.json#/Resource"
+        },
```

Comment:
Curious if SDKgen will properly handle arrays and maps. Can you add those?

Reply:
Added these to the `types_builtin` generator.